lib/dependency_validator.dart

```diff
@@ -57,6 +63,29 @@ void run() {
 
   logger.info('Validating dependencies for $packageName\n');
 
+  // `loadPackageConfig` (used below) relies on having a file built from an absolute URI.
+  final packageConfig = File.fromUri(Uri.file('${Directory.current.path}/.dart_tool/package_config.json'));
+
+  // URIs in packageConfig may be relative to its directory. Change the working directory to account for this.
+  final rootDir = Directory.current;
+  final dir = Directory('${rootDir.path}/.dart_tool');
+  Directory.current = dir;
+
+  // Find packages that provide executables.
+  final packagesWithExecutables = Set<String>();
+  for (final package in (await loadPackageConfig(packageConfig)).packages) {
+    final path = package.root.path;
+    if (Directory('$path${path.endsWith('/') ? '' : '/'}bin').existsSync()) {
```

Comment:
The [`path`](https://pub.dev/packages/path) package should be used here to join filepaths:
```dart
import 'package:path/path.dart' as p;
```

```suggestion
    if (Directory(p.join(package.root.path, 'bin')).existsSync()) {
```